File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/OrderDetailsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderDetailsRow } from "./OrderDetailsRow";

export interface OrderDetailsColumns {
    OrderCustomerId: Column<OrderDetailsRow>;
    ProductName: Column<OrderDetailsRow>;
    UnitPrice: Column<OrderDetailsRow>;
    Quantity: Column<OrderDetailsRow>;
    Discount: Column<OrderDetailsRow>;
    DetailId: Column<OrderDetailsRow>;
}

export class OrderDetailsColumns extends ColumnsBase<OrderDetailsRow> {
    static readonly columnsKey = 'Northwind.OrderDetails';
    static readonly Fields = fieldsProxy<OrderDetailsColumns>();
}

File: MovieProject/MovieProject.Web/Modules/Northwind/OrderDetails/OrderDetailsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MovieProject.Northwind.Columns;

[ColumnsScript("Northwind.OrderDetails")]
[BasedOnRow(typeof(OrderDetailsRow), CheckNames = true)]
public class OrderDetailsColumns
{
    public string OrderCustomerId { get; set; }
    public string ProductName { get; set; }
    public decimal UnitPrice { get; set; }
    public short Quantity { get; set; }
    public float Discount { get; set; }
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DetailId { get; set; }
}